org.civicrm.reqangular/src/common/decorators/angular-date/datepicker-addon.js

```diff
@@ -1,35 +1,88 @@
-define([], function () {
+/* eslint-env amd */
+
+define([
+  'common/moment'
+], function (moment) {
   'use strict';
 
   /**
    * Decorates the uib-datepicker-popup directive, making its sibling '.input-group-addon'
    * trigger the datepicker
    */
-  return ['$delegate', function ($delegate) {
+  return ['$delegate', '$compile', function ($delegate, $compile) {
     var directive = $delegate[0];
+    var isMobile = document.getElementsByTagName('body')[0].classList.contains('mobile');
```

Comment:
Could you abstract this in a separate service? Something like `detectDevice`
```js
detectDevice(); // mobile|desktop
```
this way the decorator doesn't need to know how the device is being detected